### DISCUSSION TITLE
Ignore /docs in cicd workflow

### DISCUSSION THREAD

**Discussion by msomp** [microsoft/AL-Go#1558] [Ideas]
Hi Freddy and Team,

as I'm using GitHub Pages for documentation purpose, it would be great if the complete /docs path could be ignored in cicd.
Even if the docs folder (and subfolders) mainly contains .md files (which are already ignored), there may be also other filetypes which now may trigger an unnecessary start of cicd workflow.
/docs is btw a standard GitHub folder to built GitHub pages from selected branch.

Thanks.

**Comment by freddydk** [comment]
Sounds like a good idea.
Will probably make paths-ignore configurable and add /docs to the default.

**Comment by fvet** [comment]
Will there be future AL-GO support for auto-publishing source code documentation using **aldoc.exe** as part of the release action? (Incl. publishing documentation from the private source repo to a public github (pages) repo)

References
- https://github.com/orgs/microsoft/projects/521/views/1?filterQuery=doc&pane=issue&itemId=27095580
- https://svirlan.com/business-central-aldoc-docfx-generate-github-page-to-host-your-al-documentation/

**Comment by freddydk** [comment]
Yes, working on that - https://github.com/microsoft/AL-Go/pull/760
Example: https://microsoft.github.io/bcsamples-bingmaps.pte/index.html

As you can see in the sample, it will create documentation for latest (work in progress) but also for shipped versions on the same site and you can determine how many versions you want.

Will come in preview later this month is the plan

**Comment by fvet** [comment]
Nice work, looks promising 👍

**Comment by freddydk** [comment]
Moved to roadmap

**Comment by TKapitan** [comment]
Hi @freddydk, this would help to solve one of the issues I have when assessing our move from Azure DevOps to AL Go. We have docs and a few other folders used by our internal tools that I'd like to exclude from CI/CD (they are changed quite often and won't impact BC so no need for us to run CICD).

**Comment by freddydk** [comment]
I think it makes a lot of sense to be able to control paths-ignore - will have a look at what I can do.
